refinedstorage2-core-api/src/main/java/com/refinedmods/refinedstorage2/api/core/CoreValidations.java

```diff
@@ -0,0 +1,43 @@
+package com.refinedmods.refinedstorage2.api.core;
+
+import java.util.Collection;
+import javax.annotation.Nullable;
+
+import org.apiguardian.api.API;
+
+@API(status = API.Status.STABLE, since = "2.0.0-milestone.2.0")
+public final class CoreValidations {
+    private CoreValidations() {
+    }
+
+    public static <T> T validateNotNull(@Nullable final T value, final String message) {
+        if (value == null) {
+            throw new NullPointerException(message);
+        }
+        return value;
+    }
+
+    public static void validateNonNegative(final long value, final String message) {
+        if (value < 0) {
+            throw new IllegalArgumentException(message);
```

Comment:
Should probably print the negative value, it might be quite insightful.

Reply:
Not having to format every call to a validation function was the reason I added the following function to the transfer API:
```java
	public static void notNegative(long amount) {
		if (amount < 0) {
			throw new IllegalArgumentException("Amount may not be negative, but it is: " + amount);
		}
	}
```